expression/builtin_compare.go

```diff
@@ -1070,68 +1070,91 @@ func isTemporalColumn(expr Expression) bool {
 }
 
 // tryToConvertConstantInt tries to convert a constant with other type to a int constant.
-func tryToConvertConstantInt(ctx sessionctx.Context, isUnsigned bool, con *Constant) (_ *Constant, isAlwaysFalse bool) {
+func tryToConvertConstantInt(ctx sessionctx.Context, targetFieldType *types.FieldType, con *Constant) (
+	correctVal *Constant, // The computed result. If the constant can convert to int, return the val. Else return 'con'(the input).
```

Comment:
Does it seem that we can define the output parameters as `(_ *Constant, isExceptional bool)`?